types/types.proto

```diff
@@ -0,0 +1,34 @@
+syntax = "proto3";
+
+package types;
+option go_package = "github.com/tendermint/tendermint/types";
+
+import "third_party/proto/gogoproto/gogo.proto";
+
+// BlockIdFlag indicates which BlcokID the signature is for
+enum BlockIDFlag {
+  BLOCK_ID_FLAG_UNSPECIFIED = 0;
+  BLOCK_ID_FLAG_ABSENT = 1;
+  BLOCK_ID_FLAG_COMMIT = 2;
+  BLOCK_ID_FLAG_NIL = 3;
+}
+
+// SignedMsgType is a type of signed message in the consensus.
+enum SignedMsgType {
+  SIGNED_MSG_TYPE_UNSPECIFIED = 0;
+  SIGNED_MSG_TYPE_PREVOTE_TYPE = 1;
+  SIGNED_MSG_TYPE_PRECOMMIT_TYPE = 2;
+  SIGNED_MSG_TYPE_PRPOSAL_TYPE = 3;
+}
+
+// PartsetHeader
+message PartSetHeader {
```

Comment:
which is preferable, we have .proto files for specific types (block types lives in block.proto, vote types live in vote.proto) or all in one file?

Reply:
all in one if there are not too many of them